src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/Functions/SubscriptionManagerTests.cs

```diff
@@ -0,0 +1,554 @@
+using Moq;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Notifier.Tests.Utils.NotifierTestUtils;
+using static GovUk.Education.ExploreEducationStatistics.Common.TableStorageTableNames;
+
+namespace GovUk.Education.ExploreEducationStatistics.Notifier.Tests.Functions;
+
+public class SubscriptionManagerFunctionTests(NotifierFunctionsIntegrationTestFixture fixture)
+    : NotifierFunctionsIntegrationTest(fixture)
+{
+    private static readonly GovUkNotifyOptions.EmailTemplateOptions EmailTemplateOptions = new()
+    {
+        ReleaseAmendmentPublishedId = "release-amendment-published-id",
+        ReleaseAmendmentPublishedSupersededSubscribersId = "release-amendment-published-superseded-subscribers-id",
+        ReleasePublishedId = "release-published-id",
+        ReleasePublishedSupersededSubscribersId = "release-published-superseded-subscribers-id",
+        SubscriptionConfirmationId = "subscription-confirmation-id",
+        SubscriptionVerificationId = "subscription-verification-id"
+    };
+
+    [Fact]
+    public async Task SendsSubscriptionVerificationEmail()
+    {
+        // Arrange (mocks)
+        var storageTableService = new StorageTableService(Options.Create(new AppSettingOptions
+        {
+            TableStorageConnectionString = fixture.TableStorageConnectionString()
+        }));
+
+        var notificationClient = MockNotificationClient(HttpStatusCode.OK, "response content");
+        var notificationClientProvider = new Mock<INotificationClientProvider>(MockBehavior.Strict);
+        notificationClientProvider.Setup(mock =>
+                mock.Get())
+            .Returns(notificationClient);
+
+
+        var tokenService = new Mock<ITokenService>(MockBehavior.Strict);
+        tokenService.Setup(mock =>
+                mock.GenerateToken("test1@test.com", It.IsAny<DateTime>()))
+            .Returns("activation-code-1");
+
+        var emailService = new Mock<IEmailService>(MockBehavior.Strict);
+        emailService.Setup(mock =>
+            mock.SendEmail(notificationClient, "test1@test.com", "subscription-verification-id",
+                It.IsAny<Dictionary<string, dynamic>>()));
+
+        var notifierFunction = BuildFunction(
+            storageTableService: storageTableService,
+            notificationClientProvider: notificationClientProvider.Object,
+            tokenService: tokenService.Object,
+            emailService: emailService.Object);
+
+        var request = new NewPendingSubscriptionRequest
+        {
+            Id = "test-id-1",
+            Slug = "test-publication-slug-1",
+            Email = "test1@test.com",
+            Title = "Test Publication Title 1"
+        };
+
+        // Act
+        var result =
+            await notifierFunction.RequestPendingSubscriptionFunc(request, new TestFunctionContext(),
+                new CancellationToken());
+
+        // Assert
+        var okResult = Assert.IsType<OkObjectResult>(result);
```

Comment:
Not super important, but I'd suggest that the initial type assertion here is sufficient - the null and status code checks which follow don't confirm anything new/different (i.e. an `OkObjectResult` is equivalent to a 200, and non-null is implied by the type check).

Reply:
Yeah you're 100% right, will remove